File: Quick Starts/Validation/CS/BusinessEntities/Address.cs
//===============================================================================
// Microsoft patterns & practices Enterprise Library
// Validation Application Block QuickStart
//===============================================================================
// Copyright © Microsoft Corporation.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
//===============================================================================

using System;
using System.Collections.Generic;
using System.Text;
using Microsoft.Practices.EnterpriseLibrary.Validation;
using Microsoft.Practices.EnterpriseLibrary.Validation.Validators;
using ValidationQuickStart.CustomValidators;

namespace ValidationQuickStart.BusinessEntities
{
    public class Address
    {
        private string line1;
        private string line2;
        private string city;
        private string state;
        private string postCode;
        private string country;

        [StringLengthValidator(5, 50, Ruleset = "RuleSetA")]
        public string Line1
        {
            get { return line1; }
            set { line1 = value; }
        }

        [IgnoreNulls]
        [ValidatorComposition(CompositionType.Or, Ruleset = "RuleSetA", MessageTemplate = "Address line 2 must be empty, or between 5 and 50 characters")]
        [StringLengthValidator(0, Ruleset = "RuleSetA")]
        [StringLengthValidator(5, 50, Ruleset = "RuleSetA")]
        public string Line2
        {
            get { return line2; }
            set { line2 = value; }
        }

        [StringLengthValidator(1, 50, Ruleset = "RuleSetA")]
        public string City
        {
            get { return city; }
            set { city = value; }
        }

        [USStateValidator(Ruleset = "RuleSetA")] 
        public string State
        {
            get { return state; }
            set { state = value; }
        }

        [RegexValidator(@"\d{5}(-\d{4})?", MessageTemplate="\"{0}\" is not a valid US zip code", Ruleset = "RuleSetA")]
        public string PostCode
        {
            get { return postCode; }
            set { postCode = value; }
        }

        [DomainValidator("US", "USA", "United States", "United States of America", Ruleset = "RuleSetA", MessageTemplate = "Country must be USA")]
        public string Country
        {
            get { return country; }
            set { country = value; }
        }

    }
}

File: Quick Starts/Validation/VB/BusinessEntities/Address.vb
'===============================================================================
' Microsoft patterns & practices Enterprise Library
' Validation Application Block QuickStart
'===============================================================================
' Copyright © Microsoft Corporation.  All rights reserved.
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
' OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
' LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
' FITNESS FOR A PARTICULAR PURPOSE.
'===============================================================================

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Microsoft.Practices.EnterpriseLibrary.Validation
Imports Microsoft.Practices.EnterpriseLibrary.Validation.Validators
Imports ValidationQuickStart.CustomValidators


Public Class Address

    Private _line1 As String
    Private _line2 As String
    Private _city As String
    Private _state As String
    Private _postCode As String
    Private _country As String

    <StringLengthValidator(5, 50, Ruleset:="RuleSetA")> _
    Public Property Line1() As String
        Get
            Return _line1
        End Get
        Set(ByVal value As String)
            _line1 = value
        End Set
    End Property


    <IgnoreNulls()> _
    <ValidatorComposition(CompositionType.Or, Ruleset:="RuleSetA", MessageTemplate:="Address line 2 must be empty, or between 5 and 50 characters")> _
    <StringLengthValidator(0, Ruleset:="RuleSetA")> _
    <StringLengthValidator(5, 50, Ruleset:="RuleSetA")> _
    Public Property Line2() As String
        Get
            Return _line2
        End Get
        Set(ByVal value As String)
            _line2 = value
        End Set
    End Property

    <StringLengthValidator(1, 50, Ruleset:="RuleSetA")> _
    Public Property City() As String
        Get
            Return _city
        End Get
        Set(ByVal value As String)
            _city = value
        End Set
    End Property


    <USStateValidator(Ruleset:="RuleSetA")> _
    Public Property State() As String
        Get
            Return _state
        End Get
        Set(ByVal value As String)
            _state = value
        End Set
    End Property

    <RegexValidator("\d{5}(-\d{4})?", MessageTemplate:="""{0}"" is not a valid US zip code", Ruleset:="RuleSetA")> _
    Public Property PostCode() As String
        Get
            Return _postCode
        End Get
        Set(ByVal value As String)
            _postCode = value
        End Set
    End Property


    <DomainValidator("US", "USA", "United States", "United States of America", Ruleset:="RuleSetA", MessageTemplate:="Country must be USA")> _
    Public Property Country() As String
        Get
            Return _country
        End Get
        Set(ByVal value As String)
            _country = value
        End Set
    End Property


End Class
